test/junit/scala/collection/ParallelConsistencyTest.scala

```diff
@@ -0,0 +1,45 @@
+package scala.collection.immutable
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(classOf[JUnit4])
+class ParallelConsistencyTest {
+
+  private val sensibleSeq = Seq(1,2,3)
+
+  /** Tests to make sure .seq is used:
+    *   if it is, elements are 1, 2, 3
+    *   if not, returns 2, 3, 1 with apply or 3, 2, 1 with foreach
+    */
+  private val pathologicalGenSeq: collection.GenSeq[Int] = new collection.parallel.ParSeq[Int] {
+    def length = 3
+    
+    // This methods is surely sequential & safe
+    def seq = sensibleSeq
+    
+    // These methods could be used dangerously either directly or indirectly, should never call them
+    def apply(i: Int) = seq((i+1)%3)
```

Comment:
It would be correct to implement using something like:

```
for (i <- 0 to genSeq.length) builder += genSeq(i)
```

I think the clearest way to communicate the intent of the test will be to throw an exception from `apply` and `foreach`, and add a comment to the effect of "all access must be through .seq`.